src/PowerShellEditorServices.Hosting/EditorServicesConfig.cs

```diff
@@ -0,0 +1,134 @@
+        /// with all required fields.
+        /// </summary>
+        /// <param name="hostInfo">The host description object.</param>
+        /// <param name="psHost">The PowerShell host to use in Editor Services.</param>
+        /// <param name="sessionDetailsPath">The path to use for the session details file.</param>
+        /// <param name="bundledModulePath">The path to the modules bundled with Editor Services.</param>
+        /// <param name="logPath">The path to be used for Editor Services' logging.</param>
+        public EditorServicesConfig(
+            HostInfo hostInfo,
+            PSHost psHost,
+            string sessionDetailsPath,
+            string bundledModulePath,
+            string logPath)
+        {
+            HostInfo = hostInfo;
+            PSHost = psHost;
+            SessionDetailsPath = sessionDetailsPath;
+            BundledModulePath = bundledModulePath;
+            LogPath = logPath;
+        }
+
+        /// <summary>
+        /// The host description object.
+        /// </summary>
+        public HostInfo HostInfo { get; }
+
+        /// <summary>
+        /// The PowerShell host used by Editor Services.
+        /// </summary>
+        public PSHost PSHost { get; }
+
+        /// <summary>
+        /// The path to use for the session details file.
+        /// </summary>
+        public string SessionDetailsPath { get; }
+
+        /// <summary>
+        /// The path to the modules bundled with EditorServices.
+        /// </summary>
+        public string BundledModulePath { get; }
+
+        /// <summary>
+        /// The path to use for logging for Editor Services.
+        /// </summary>
+        public string LogPath { get; }
+
+        /// <summary>
+        /// Names of or paths to any additional modules to load on startup.
+        /// </summary>
+        public IReadOnlyList<string> AdditionalModules { get; set; } = null;
+
+        /// <summary>
+        /// Flags of features to enable on startup.
+        /// </summary>
+        public IReadOnlyList<string> FeatureFlags { get; set; } = null;
+
+        /// <summary>
+        /// The console REPL experience to use in the integrated console
+        /// (including none to disable the integrated console).
+        /// </summary>
+        public ConsoleReplKind ConsoleRepl { get; set; } = ConsoleReplKind.None;
+
+        /// <summary>
+        /// The minimum log level to log events with.
+        /// </summary>
+        public PsesLogLevel LogLevel { get; set; } = PsesLogLevel.Normal;
+
+        /// <summary>
+        /// Configuration for the language server protocol transport to use.
+        /// </summary>
+        public ITransportConfig LanguageServiceTransport { get; set; } = null;
+
+        /// <summary>
+        /// Configuration for the debug adapter protocol transport to use.
+        /// </summary>
+        public ITransportConfig DebugServiceTransport { get; set; } = null;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this initialization to 'DebugServiceTransport', the compiler will do that for you.](https://app.codacy.com/app/TylerLeonhardt/PowerShellEditorServices/pullRequest?prid=4579423)